### ISSUE TITLE
Code generation fails for method writer if target interface extends Marshallable

### ISSUE THREAD

**Issue by glukos** [OpenHFT/Chronicle-Wire#205]
Example of generated code for method WriteMarshallable#writeValue:
```
    public void writeValue(final ValueOut out) {
        try (final DocumentContext dc = this.out.get().acquireWritingDocument(false)) {
            if (out.get().recordHistory()) MessageHistory.writeHistory(dc);
            final ValueOut valueOut = dc.wire().writeEventName("writeValue");
            valueOut.object(net.openhft.chronicle.wire.ValueOut.class, out);
        }
    }
```
Its compilation fails as `out` in `out.get().recordHistory()` clashes with a parameter:
```
/net/openhft/chronicle/queue/impl/single/MessageHistoryTestMyInterface2BinarylightMethodWriter.java:112: error: cannot find symbol
            if (out.get().recordHistory()) MessageHistory.writeHistory(dc);
                   ^
  symbol:   method get()
  location: variable out of type net.openhft.chronicle.wire.ValueOut
```
Disambiguation through `this.out` should resolve the issue.

**Comment by peter-lawrey** [comment]
We should have a check to prevent this. It is assumed that marshallable and packaged in method calls, and the interface itself shouldn't be Marshallable.

**Comment by hft-team-city** [comment]
Released in [Chronicle-Wire-2.20.41](https://github.com/OpenHFT/Chronicle-Wire/releases/chronicle-wire-2.20.41), [BOM-2.20.67](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.20.67)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Wire-2.20.101](https://github.com/OpenHFT/Chronicle-Wire/releases/chronicle-wire-2.20.101), [BOM-2.20.134](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.20.134)
